odbc/src/api/functions.rs

```diff
@@ -832,7 +831,7 @@ pub unsafe extern "C" fn SQLDisconnect(connection_handle: HDbc) -> SqlReturn {
 
 fn sql_driver_connect(conn: &Connection, odbc_uri_string: &str) -> Result<MongoConnection> {
     let mut odbc_uri = ODBCUri::new(odbc_uri_string)?;
-    let mongo_uri = odbc_uri.remove_to_mongo_uri()?;
+    let client_options = odbc_uri.try_into_client_options()?;
     let auth_src = odbc_uri.remove_or_else(|| "admin", &["auth_src"]);
```

Comment:
I can probably move this auth_src part into the try_into_client_options, also.

Reply:
Yes, plus I don't think we'll need it in connect once we move from using a `ping` command to using a `Select 1` query to validate the connection.